robotPositioner/robotPositioner.h

```diff
@@ -0,0 +1,382 @@
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+    * reads in the coordinates from a CSV file and puts them in a list
+    */
+    void readCoordinatesFromFile(const std::string csv_file) {
+        std::ifstream in(csv_file);
+        std::vector<std::vector<double>> fields;
+        if (in) {
+            std::string line;
+            while (getline(in, line)) {
+                std::stringstream sep(line);
+                std::string field;
+                fields.push_back(std::vector<double>());
+                while (getline(sep, field, ',')) {
+                    fields.back().push_back(stof(field));
+                }
+            }
+        }
+
+        // saving coordinates from csv
+        for (auto row : fields) {	    	
+            std::vector<double> temp_vec;
+            for (double field : row) {
+                temp_vec.push_back(field);         	
+            }
+            m_coordinates.push_back(temp_vec);
+        }
+    }
+
+    // drives the robot the desired pose (x,y,angle)
+    void driveRobotToGoalLocation() {
+        bool stopFlag = true;
+
+        while (stopFlag) {
+            // updates the range and bearing
+            updateRangeAndBearing();
+
+            // orientation of Target with respect to the line of sight from the observer to the target
+            double theta = m_heading + m_bearingFromGoal - m_goalAngle; 
+            theta = angleWrapAround(theta);
+
+            // curveness value: k
+            double k = -(1/m_distanceFromGoal)* ( m_k2* (m_bearingFromGoal- atan(-m_k1*theta*PI/180.0)*180.0/PI) + 
+                1+(m_k1/(1+pow(m_k1*theta,2)))*sin(m_bearingFromGoal*PI/180.0)*180.0/PI);
+
+            // current velocity
+            double v = m_max_velocity/(1+m_beta*pow(abs(k),m_alpha));
+
+            // formula for curvy path
+            double w = k*v;
+       
+            // if we are close to the goal and the heading error is less than the allowed, stop the robot
+            if (m_distanceFromGoal < m_stopping_distance && closeTo(m_bearingFromGoal,m_allowed_heading_error)) {
+
+                stopFlag = false;
+                m_bot->tank(0,0);
+            } else {
+                driveRobotWithVelocities(v,w, m_robot_r, m_robot_D);
+            }
+
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        }
+    }
+
+    // drives the robot wheels using v and w as velocities which get translated to 
+    // indiviual wheel velocities. Conversion from these velocities to individual 
+    // wheel velocities is done via solving simultaneous equations
+    // parameters:
+    //             v      : translational velocity
+    //             w      : rotational velocity
+    //             robot_r: wheel radius
+    //             robot_d: axis length
+    template <class T, class S>
+    void driveRobotWithVelocities(const T v, const T w, const S robot_r, const S robot_d) {
```

Comment:
Not sure why this function is templated?

The units lib would be useful here so you can have unit types as parameters; as it is it's not clear what the units of velocity and distance etc. are